### PR TITLE
bug; aborted workflows bubble up the error and stay in running

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
Abort can be initiated in 2 cases. Abort method now handles both the cases.
Case 1: User initiated. All retries are taken up
Case 2: System failure, atleast one retry is now available

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
NA

## Tracking Issue
https://github.com/lyft/flyte/issues/309

## Follow-up issue
NA

### CODE REVIEW

File: pkg/controller/workflow/executor.go [hunk 1]
```diff
@@ -423,12 +423,12 @@ func (c *workflowExecutor) HandleAbortedWorkflow(ctx context.Context, w *v1alpha
 		if w.Status.FailedAttempts > maxRetries {
 			reason = fmt.Sprintf("max number of system retry attempts [%d/%d] exhausted - system failure.", w.Status.FailedAttempts, maxRetries)
 			err = errors.Errorf(errors.RuntimeExecutionError, w.GetID(), "max number of system retry attempts [%d/%d] exhausted. Last known status message: %v", w.Status.FailedAttempts, maxRetries, w.Status.Message)
-		}
-
-		// Best effort clean-up.
-		if err2 := c.cleanupRunningNodes(ctx, w, reason); err2 != nil {
-			logger.Errorf(ctx, "Failed to propagate Abort for workflow:%v. Error: %v", w.ExecutionID.WorkflowExecutionIdentifier, err2)
-			return err2
+		} else {
```
Comment (EngHabu) [inline pkg/controller/workflow/executor.go:426]: aaah

File: pkg/controller/workflow/executor.go [hunk 2]
```diff
@@ -417,18 +417,24 @@ func (c *workflowExecutor) HandleAbortedWorkflow(ctx context.Context, w *v1alpha
...
-		reason := "User initiated workflow abort."
+		reason := fmt.Sprintf("max number of system retry attempts [%d/%d] exhausted - system failure.", w.Status.FailedAttempts, maxRetries)
 		c.metrics.IncompleteWorkflowAborted.Inc(ctx)
-		var err error
-		if w.Status.FailedAttempts > maxRetries {
-			reason = fmt.Sprintf("max number of system retry attempts [%d/%d] exhausted - system failure.", w.Status.FailedAttempts, maxRetries)
-			err = errors.Errorf(errors.RuntimeExecutionError, w.GetID(), "max number of system retry attempts [%d/%d] exhausted. Last known status message: %v", w.Status.FailedAttempts, maxRetries, w.Status.Message)
+		// Check of the workflow was deleted and that caused the abort
+		if w.GetDeletionTimestamp() != nil {
+			reason = "User initiated workflow abort."
```
Comment (EngHabu) [inline pkg/controller/workflow/executor.go:424]: Since this might not always be "user" initiated, maybe we should reword this to "Workflow aborted" to avoid confusion until we plumb through the reason through propeller & admin.

Comment (kumare3) [reply pkg/controller/workflow/executor.go]: done
